### PR TITLE
Bump phpstan to level 5

### LINKED ISSUE #296: Bump phpstan to level 5
### Feature description

Bump phpstan to level 5

### PR DESCRIPTION
**Pull request by blankse**
Fix https://github.com/pimcore/customer-data-framework/issues/296

### CODE REVIEW

**Comment by dvesh3** [comment]
@blankse  please rebase to fix the failing Codeception test.

**Comment by blankse** [comment]
@dvesh3 Done

**Comment by dvesh3** [comment]
@blankse please confirm if you have these changes on your branch https://github.com/pimcore/customer-data-framework/commit/8c6aa5a3857a4635c3027dc7178f79e00efeb890

**Comment by blankse** [comment]
@dvesh3 The commit is in the branch. See https://github.com/blankse/customer-data-framework/commit/8c6aa5a3857a4635c3027dc7178f79e00efeb890

**Comment by blankse** [comment]
@dvesh3 With new ramsey/composer-install version it seems to work.

**Comment by dvesh3** [comment]
> @dvesh3 With new ramsey/composer-install version it seems to work.

Interesting!! thanks for your fix. I will start reviewing your changes then 😉

**Comment by dvesh3** [comment]
@blankse Other PR is merged to bump composer-install, please rebase again. thanks!

**Comment by blankse** [comment]
@dvesh3 Done

File: src/Model/CustomerView/FilterDefinition.php [hunk 1]
```diff
@@ -48,10 +48,10 @@ class FilterDefinition extends AbstractModel
     /** @var bool */
     private $shortcutAvailable = false;
 
-    /** @var int|null */
+    /** @var string|null */
```
Comment (dvesh3) [inline src/Model/CustomerView/FilterDefinition.php:51]: why string? column type is `bigint(20)`

File: src/Model/CustomerView/FilterDefinition.php [hunk 2]
```diff
@@ -48,10 +48,10 @@ class FilterDefinition extends AbstractModel
...
     private $creationDate = null;
 
-    /** @var int|null */
+    /** @var string|null */
```
Comment (dvesh3) [inline src/Model/CustomerView/FilterDefinition.php:54]: why string? column type is bigint(20)

File: src/SegmentManager/SegmentManagerInterface.php [hunk 3]
```diff
@@ -130,7 +131,7 @@ public function getSegmentBuilders();
      *    Therefore it's not a good idea to rely on the execution timestamp (time()) here.
      *    It's necessary to do this based on other criterias (for example activities).
      *  - If null was passed the timestamp will be set to null
-     * @param int|true|null $segmentApplicationCounter
+     * @param int|bool|null $segmentApplicationCounter
```
Comment (dvesh3) [inline src/SegmentManager/SegmentManagerInterface.php:134]: I guess this should stay `true` as null is passed when not set.

File: src/Model/CustomerView/FilterDefinition.php (hunk 1 above)
Comment (blankse) [reply src/Model/CustomerView/FilterDefinition.php:51]: @dvesh3 Is set to string here https://github.com/pimcore/customer-data-framework/blob/59753f6dd7e1b98f40c9d4350cfbbc792d557d49/src/Model/CustomerView/FilterDefinition/Dao.php#L100-L104 and the column type is `DATETIME` see https://github.com/pimcore/customer-data-framework/blob/59753f6dd7e1b98f40c9d4350cfbbc792d557d49/src/Resources/sql/filterDefinition/datamodel.sql#L9-L10

File: src/SegmentManager/SegmentManagerInterface.php (hunk 3 above)
Comment (blankse) [reply src/SegmentManager/SegmentManagerInterface.php:134]: @dvesh3 Can be false. Is set to false here https://github.com/pimcore/customer-data-framework/blob/59753f6dd7e1b98f40c9d4350cfbbc792d557d49/src/ActionTrigger/Action/AddTrackedSegment.php#L86

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@blankse thank you!

**Pull request merged**